From 2fc28d5eb501febaab1b3e94239b262291bcba53 Mon Sep 17 00:00:00 2001
From: "Yaoming.Qin" <B10945@freescale.com>
Date: Fri, 13 May 2011 18:40:10 +0800
Subject: [PATCH 144/635] ENGR00143382  The camera's taking picure has wrong
 rotation

The issue is because that the rotate is set as the angle, which is
mistaken before.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 libcamera/CameraHal.cpp | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 643c5b8..4936939 100755
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -928,7 +928,7 @@ Pic_out:
         struct jpeg_enc_datetime_info_t datetime_info;
         struct jpeg_enc_focallength_t focallength_info;
         struct jpeg_enc_gps_param gps_info;
-        JPEG_ENCODER_ROTATION rotate_info;
+        int rotate_angle = 0;
         JPEG_ENCODER_WHITEBALANCE whitebalance_info;
         JPEG_ENCODER_FLASH flash_info;
         const char * pWhiteBalanceStr, *pFlashStr;
@@ -975,9 +975,18 @@ Pic_out:
         memcpy((char *)datetime_info.datetime, temp_string, sizeof(datetime_info.datetime));
         mJpegEncCfg.pDatetimeInfo = &datetime_info;
 
-        rotate_info = (JPEG_ENCODER_ROTATION)mParameters.getInt(CameraParameters::KEY_ROTATION);
-        mJpegEncCfg.RotationInfo = rotate_info; //the android and the jpeg has the same define
-        CAMERA_HAL_LOG_INFO("ratate info is %d", rotate_info);
+        rotate_angle = mParameters.getInt(CameraParameters::KEY_ROTATION);
+        if (rotate_angle == 0)
+            mJpegEncCfg.RotationInfo = ORIENTATION_NORMAL; //the android and the jpeg has the same define
+        else if (rotate_angle == 90)
+            mJpegEncCfg.RotationInfo = ORIENTATION_ROTATE_90;
+        else if (rotate_angle = 180)
+            mJpegEncCfg.RotationInfo = ORIENTATION_ROTATE_180;
+        else if (rotate_angle = 270)
+            mJpegEncCfg.RotationInfo = ORIENTATION_ROTATE_270;
+        else
+            mJpegEncCfg.RotationInfo = ORIENTATION_NORMAL;
+        CAMERA_HAL_LOG_INFO("ratate info is %d", rotate_angle);
 
         pWhiteBalanceStr = mParameters.get(CameraParameters::KEY_WHITE_BALANCE);
         CAMERA_HAL_LOG_INFO("white balance is %s",pWhiteBalanceStr);
-- 
1.8.0

